### PR TITLE
Add feedback form

### PR DESCRIPTION
**Pull request by nichochar**
After trying to use airtable and getting very annoyed at how clunky it is, I figured out we could simply use google sheets with their "app scripts" API. This is free, and works.

### Note
* I realized that posthog has a "surveys" product. It has some nice things (like NPS score or PMF questionnaire), but as far as I could tell, you have to use their UI, so it's styled inconsistently with the rest of the app, and it says "by posthog" in the middle of it. Holding off of that for now, but good to consider if we want to do more advanced surveys later.

CleanShot 2024-07-27 at 16 14 15@2x

### CODE REVIEW

File: packages/web/src/lib/utils.ts [hunk 1]
```diff
@@ -27,3 +27,18 @@ export function getTitleForSession(session: SessionType) {
   const titleCell = session.cells.find((cell: CellType) => cell.type === 'title') as TitleCellType;
   return titleCell?.text;
 }
+
+// POST feedback to Google Sheets
+// Because of CORS, we don't get a response here so this is a fire and forget.
```
Comment (benjreinhart) [inline packages/web/src/lib/utils.ts:32]: Why not make this request from the API where CORS is not an issue?

File: packages/web/src/components/feedback-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const [feedback, setFeedback] = useState('');
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <DialogContent className="max-w-xl">
+        <DialogHeader>
+          <DialogTitle>Share Feedback</DialogTitle>
+          <DialogDescription asChild>
+            <div className="pt-4 flex flex-col gap-2">
+              <p>
+                We're always looking to improve Srcbook, and your feedback is invaluable.
```
Comment (benjreinhart) [inline packages/web/src/components/feedback-dialog.tsx:30]: There shouldn't be a comma after the "and"

**Comment by whatrocks** [comment]
I'd suggest optionally letting the user leave their email so that you can reply to them

File: packages/web/src/lib/utils.ts (hunk 1 above)
Comment (nichochar) [reply packages/web/src/lib/utils.ts]: Duh, I was clearly tired as I did not think of this. Fixed.

**Comment by nichochar** [comment]
> I'd suggest optionally letting the user leave their email so that you can reply to them

Good suggestion @whatrocks , added

**Pull request merged**